### PR TITLE
journal_server: check for writership before creating journal

### PR DESCRIPTION
**Pull request by strib**
So we don't create worthless on-disk journals for public TLFs, etc.

Issue: KBFS-2643

### CODE REVIEW

File: libkbfs/journal_server_test.go [hunk 1]
```diff
@@ -747,6 +748,45 @@ func TestJournalServerEnableAuto(t *testing.T) {
...
+		ctx, config.KBPKI(), nil, "test_user2", tlf.Public)
+	require.NoError(t, err)
+
+	// Access someone else's public TLF, which should NOT create a
+	// journal automatically.
+	_, err = config.MDOps().GetIDForHandle(ctx, h)
+	require.NoError(t, err)
+
+	status, tlfIDs = jServer.Status(ctx)
+	require.True(t, status.EnableAuto)
```
Comment (jzila) [inline libkbfs/journal_server_test.go]: What is `EnableAuto`?

Comment (strib) [reply libkbfs/journal_server_test.go]: `EnableAuto` means that journals are created automatically when you access them (now the default for everyone, but it didn't used to be).

Comment (jzila) [reply libkbfs/journal_server_test.go]: All right, so this check is essentially to make sure that the `EnableAuto=false` switch isn't the cause `status.JournalCount==0`.

Comment (strib) [reply libkbfs/journal_server_test.go]: Right.

**Pull request merged**